src/hooks/useAvailableRepositories.js

```diff
@@ -1,33 +1,49 @@
 import { useQuery } from 'react-query';
 import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
 
-const fetchAdditionalRepositories = (token) => async (keyName) => {
+const fetchAdditionalRepositories = async (
+  token,
+  keyName,
+  offset = 0,
+  allRepositories = []
+) => {
   if (!keyName) {
     return false;
   }
 
   const response = await fetch(
-    `/api/rhsm/v2/activation_keys/${keyName}/available_repositories?default=Disabled`,
+    `/api/rhsm/v2/activation_keys/${keyName}/available_repositories?default=Disabled&offset=${offset}`,
     {
       headers: { Authorization: `Bearer ${await token}` },
     }
   );
 
-  const repositoriesData = await response.json();
-
-  return repositoriesData.body;
-};
+  if (!response.ok) {
+    throw new Error('Failed to fetch repositories');
+  }
 
-const getAvailableRepositories = (token) => async (keyName) => {
-  const repositories = await fetchAdditionalRepositories(token)(keyName);
-  return repositories;
+  const repositoriesData = await response.json();
+  const repositories = repositoriesData.body;
+
+  if (repositories.length === 0 || repositories.length < 100) {
```

Comment:
```suggestion
  if (repositories.length < 100) {
```
That first check is always true if the second is